src/app/features/course-info/course-info.component.ts

```diff
@@ -1,10 +1,44 @@
-import { Component } from '@angular/core';
-
+import { Component, Input, OnInit } from "@angular/core";
+import { CoursesService } from "@app/services/courses.service";
+interface Course {
+  id: string;
+  title: string;
+  description: string;
+  creationDate: string;
+  duration: number;
+  authors: string[];
+}
 @Component({
-  selector: 'app-course-info',
-  templateUrl: './course-info.component.html',
-  styleUrls: ['./course-info.component.scss']
+  selector: "app-course-info",
+  templateUrl: "./course-info.component.html",
+  styleUrls: ["./course-info.component.scss"],
 })
-export class CourseInfoComponent {
+export class CourseInfoComponent implements OnInit {
+  @Input() course!: Course;
   // Use the names for the input `course`.
+  id: string = "";
+  title: string = "";
+  description: string = "";
+  creationDate: string = "";
+  duration: number = 0;
+  authors: string[] = [];
+
+  constructor(private coursesService: CoursesService) {}
+
+  ngOnInit(): void {
+    this.id = this.course.id;
+    this.title = this.course.title;
+    this.description = this.course.description;
+    //possible future formats
```

Comment:
You can delete this, too. Also, formatting the creationDate and displaying the course duration in a readable way are both a job for pipes. You can check those declared in /shared/pipes folder.